zebra-state/src/service.rs

```diff
@@ -720,28 +720,31 @@ impl StateService {
             return rsp_rx;
         }
 
-        if self.finalized_block_write_sender.is_none() {
-            // Wait until block commit task is ready to write non-finalized blocks before dequeuing them
-            self.send_ready_non_finalized_queued(parent_hash);
+        self.send_ready_non_finalized_queued(parent_hash);
 
-            let finalized_tip_height = self.read_service.db.finalized_tip_height().expect(
+        rsp_rx
+    }
+
+    /// Returns `true` if `hash` is a valid previous block hash for new non-finalized blocks.
+    fn can_fork_chain_at(&mut self, hash: &block::Hash) -> bool {
+        // Wait until:
+        // - we've sent the finalized block for the last checkpoint, and
+        // - it has been successfully written to disk.
+        if self.finalized_block_write_sender.is_some() {
+            return false;
+        }
+
+        let (finalized_tip_height, finalized_tip_hash) = self.read_service.db.tip().expect(
             "Finalized state must have at least one block before committing non-finalized state",
         );
 
-            self.queued_non_finalized_blocks
-                .prune_by_height(finalized_tip_height);
-
-            self.sent_non_finalized_block_hashes
-                .prune_by_height(finalized_tip_height);
-        }
+        self.queued_non_finalized_blocks
+            .prune_by_height(finalized_tip_height);
 
-        rsp_rx
-    }
+        self.sent_non_finalized_block_hashes
+            .prune_by_height(finalized_tip_height);
 
-    /// Returns `true` if `hash` is a valid previous block hash for new non-finalized blocks.
-    fn can_fork_chain_at(&self, hash: &block::Hash) -> bool {
-        self.sent_non_finalized_block_hashes.contains(hash)
-            || &self.read_service.db.finalized_tip_hash() == hash
+        self.sent_non_finalized_block_hashes.contains(hash) || &finalized_tip_hash == hash
```

Comment:
This field name is now outdated for two reasons:
1. It uses old verification status terminology
2. The type contains both checkpointed and semantically verified blocks

Reply:
> 1. It uses old verification status terminology

Let's do these name changes automatically in ticket #6793?
Any complicated documentation changes can be done in this PR, or added to the list in #6681.

Reply:
Should this PR still update this field name?